server/src/frontend_api.rs

```diff
@@ -161,24 +153,24 @@ async fn get_enabled_proxy(
 #[get("/frontend")]
 async fn get_enabled_frontend(
     edge_token: EdgeToken,
-    features_source: web::Data<dyn EdgeSource>,
+    engine_cache: web::Data<DashMap<String, EngineState>>,
     context: web::Query<Context>,
 ) -> EdgeJsonResult<FrontendResult> {
-    get_enabled_features(edge_token, features_source, context).await
+    get_enabled_features(edge_token, engine_cache, context)
 }
-async fn get_enabled_features(
+
+fn get_enabled_features(
     edge_token: EdgeToken,
-    features_source: web::Data<dyn EdgeSource>,
+    engine_cache: web::Data<DashMap<String, EngineState>>,
     context: web::Query<Context>,
 ) -> EdgeJsonResult<FrontendResult> {
-    let client_features = features_source.get_client_features(&edge_token).await;
     let context = context.into_inner();
-
-    let toggles: Vec<EvaluatedToggle> = resolve_frontend_features(client_features?, context)
-        .filter(|toggle| toggle.enabled)
-        .collect();
-
-    Ok(Json(FrontendResult { toggles }))
+    let key = crate::tokens::cache_key(edge_token);
+    let engine = engine_cache
+        .get(&key)
+        .ok_or_else(|| EdgeError::PersistenceError("Could not find data for token".into()))?;
```

Comment:
Definitely, definitely not a 200 with an empty list. This will be picked up by the SDKs and cause them to promptly eject their own memory and persistent caches, which means a 200 here will change the way features are resolved on the client SDK when they may have a healthy cache at the moment